inbox/util/concurrency.py

```diff
@@ -65,8 +71,17 @@ def wrapped(*args, **kwargs):
 def retry_with_logging(func, logger=None, retry_classes=None,
                        fail_classes=None, account_id=None, provider=None,
                        backoff_delay=BACKOFF_DELAY):
-    def callback():
-        log_uncaught_errors(logger, account_id=account_id, provider=provider)
+    network_errs = [0]
+
+    def callback(e):
+        if isinstance(e, TRANSIENT_NETWORK_ERRS):
+            network_errs[0] += 1
+            if network_errs[0] < 20:
+                return
+        else:
+            network_errs[0] = 0
+        log_uncaught_errors(logger, account_id=account_id, provider=provider,
```

Comment:
What do you think about logging the error count here in the log statement, so it's obvious to the person reading the logs that this has happened 20+ times?